src/main/java/com/gooddata/GoodData.java

```diff
@@ -192,10 +200,13 @@ protected GoodData(String hostname, String login, String password, int port, Str
         featureFlagService = new FeatureFlagService(restTemplate);
     }
 
-    private RestTemplate createRestTemplate(String hostname, HttpClient httpClient, int port, String protocol) {
-
+    static RestTemplate createRestTemplate(GoodDataEndpoint endpoint, HttpClient httpClient) {
         final UriPrefixingClientHttpRequestFactory factory = new UriPrefixingClientHttpRequestFactory(
-                new HttpComponentsClientHttpRequestFactory(httpClient), hostname, port, protocol);
+                new HttpComponentsClientHttpRequestFactory(httpClient),
+                endpoint.getHostname(),
```

Comment:
@PetrVales I understand that hostname, port and protocol are validate by endpoint, but @martiner probably refers to arguments of `createRestTemplate`